src/Avalonia.Base/Media/RenderOptions.cs

```diff
@@ -1,36 +1,181 @@
+        public static BitmapInterpolationMode GetBitmapInterpolationMode(Visual element)
         {
-            return element.GetValue(BitmapInterpolationModeProperty);
+            return element.RenderOptions.BitmapInterpolationMode;
         }
 
         /// <summary>
         /// Sets the value of the BitmapInterpolationMode attached property for a control.
         /// </summary>
         /// <param name="element">The control.</param>
         /// <param name="value">The left value.</param>
-        public static void SetBitmapInterpolationMode(AvaloniaObject element, BitmapInterpolationMode value)
+        public static void SetBitmapInterpolationMode(Visual element, BitmapInterpolationMode value)
+        {
+            element.RenderOptions = element.RenderOptions.WithBitmapInterpolationMode(value);
+        }
+
+        /// <summary>
+        /// Gets the value of the EdgeMode attached property for a control.
+        /// </summary>
+        /// <param name="element">The control.</param>
+        /// <returns>The control's left coordinate.</returns>
+        public static EdgeMode GetEdgeMode(Visual element)
+        {
+            return element.RenderOptions.EdgeMode;
+        }
+
+        /// <summary>
+        /// Sets the value of the EdgeMode attached property for a control.
+        /// </summary>
+        /// <param name="element">The control.</param>
+        /// <param name="value">The left value.</param>
+        public static void SetEdgeMode(Visual element, EdgeMode value)
+        {
+            element.RenderOptions = element.RenderOptions.WithEdgeMode(value);
+        }
+
+        /// <summary>
+        /// Gets the value of the TextRenderingMode attached property for a control.
+        /// </summary>
+        /// <param name="element">The control.</param>
+        /// <returns>The control's left coordinate.</returns>
+        public static TextRenderingMode GetTextRenderingMode(Visual element)
+        {
+            return element.RenderOptions.TextRenderingMode;
+        }
+
+        /// <summary>
+        /// Sets the value of the TextRenderingMode attached property for a control.
+        /// </summary>
+        /// <param name="element">The control.</param>
+        /// <param name="value">The left value.</param>
+        public static void SetTextRenderingMode(Visual element, TextRenderingMode value)
+        {
+            element.RenderOptions = element.RenderOptions.WithTextRenderingMode(value);
+        }
+
+        public RenderOptions WithBitmapInterpolationMode(BitmapInterpolationMode bitmapInterpolationMode)
+        {
+            return new RenderOptions
+            {
+                BitmapInterpolationMode = bitmapInterpolationMode,
+                EdgeMode = EdgeMode,
+                TextRenderingMode= TextRenderingMode
+            };
+        }
+
+        public RenderOptions WithBitmapBlendingMode(BitmapBlendingMode bitmapBlendingMode)
```

Comment:
I'm not quite sure that it belongs to RenderOptions, since it might work in a similar manner as bitmap effects